docs/index.rst

```diff
@@ -12,6 +7,117 @@ Contents:
    :maxdepth: 2
 
 
+Overview
+--------
+
+Format
+~~~~~~
+
+All response bodies are sent as JSON objects.
+
+.. TODO example success response once we have an API endpoint
+
+Where specified, query string parameters can be provided.
+
+.. TODO example request once we have an API endpoint that accepts parameters
+
+Request bodies for ``POST``, ``PUT``, ``PATCH`` and ``DELETE`` requests are expected to be JSON objects.
+
+.. TODO example request once we have an API endpoint
+
+Verbs
+~~~~~
+
+Unless otherwise specified, the following verbs are used for requests:
+
++--------+------------------------------------------------------------+
+| Verb   | Description                                                |
++========+============================================================+
+| GET    | Used for retrieving a resource                             |
++--------+------------------------------------------------------------+
+| POST   | Used for creating a resource                               |
++--------+------------------------------------------------------------+
+| PUT    | Used to replace a resouce                                  |
++--------+------------------------------------------------------------+
+| PATCH  | Used to partially update a resource with fields given in   |
+|        | the request body                                           |
++--------+------------------------------------------------------------+
+| DELETE | Used to delete a resource                                  |
++--------+------------------------------------------------------------+
+
+Client errors
+~~~~~~~~~~~~~
+
+All client error responses contain a ``type`, ``message`` and ``details``
+fields.  ``type`` is a programmitically-usable string representing the type of
+the error, ``message`` is a human-readable string describing the error and
+``details`` is an object containing data specific to the error type.
+
+.. sourcecode:: http
+
+   HTTP/1.1 400 Bad Request
+
+   {
+     "type": "parse_error",
+     "message": "Invalid JSON in request body"
+   }
+
+
+Two client errors common to API endpoints expecting a request body are **parsing errors** and **validation errors**.
+
+Parsing errors
+**************
+
+If the data provided in the request body is invalid JSON, a ``400 Bad Request`` response will be given:
+
+.. sourcecode:: http
+
+   HTTP/1.1 400 Bad Request
+   Content-Type: application/json
+
+   {
+     "type": "parse_error",
+     "message": "Invalid JSON in request body"
```

Comment:
Hmm, seems more worth adding to the message rather than putting in details. Will add it to the message.